src/extension/background-script/actions/allowances/update.ts

```diff
@@ -5,20 +5,25 @@ type OptionalPick<T, K extends keyof T> = { [P in K]?: T[P] };
 
 const updateAllowance = async (message: MessageAllowanceUpdate) => {
   const id = message.args.id;
+  if (!id) return { error: "id is missing" };
 
   const update: OptionalPick<
     Allowance,
     "totalBudget" | "remainingBudget" | "enabled" | "lnurlAuth"
   > = {};
-  if (Object.prototype.hasOwnProperty.call(message.args, "totalBudget")) {
+
+  if (message.args.totalBudget) {
     update.totalBudget = message.args.totalBudget;
     update.remainingBudget = message.args.totalBudget;
   }
-  if (Object.prototype.hasOwnProperty.call(message.args, "enabled")) {
+
+  if (message.args.enabled) {
```

Comment:
this breaks things. there is a reason why `Object.prototype.hasOwnProperty`
what values can `enabled` or `lnurlAuth` have?
try setting it to false.

Reply:
Yes, thanks! Corrected and added a test